system-test/dns.js

```diff
@@ -0,0 +1,203 @@
+        };
+
+        dns.createZone(options, function(err, zone) {
+          assert.ifError(err);
+          ZONE = zone;
+          done();
+        });
+      });
+    });
+  });
+
+  after(function(done) {
+    ZONE.delete({ force: true }, done);
+  });
+
+  it('should create a zone', function(done) {
+    var tempName = 'test-zone-' + uuid.v4().substr(0, 18);
+
+    var options = {
+      name: tempName,
+      dnsName: DNS_DOMAIN
+    };
+
+    dns.createZone(options, function(err, zone) {
+      assert.ifError(err);
+      assert.equal(zone.name, tempName);
+      zone.delete(done);
+    });
+  });
+
+  it('should return 0 or more zones', function(done) {
+    dns.getZones(function(err, zones) {
+      assert.ifError(err);
+      assert(zones.length >= 0);
+      done();
+    });
+  });
+
+  describe('Zones', function() {
+    it('should get the metadata for a zone', function(done) {
+      ZONE.getMetadata(function(err, metadata) {
+        assert.ifError(err);
+        assert.equal(metadata.name, ZONENAME);
+        done();
+      });
+    });
+
+    it('should delete a zone', function(done) {
+      var options = {
+        name: 'test-zone-' + uuid.v4().substr(0, 18),
+        dnsName: DNS_DOMAIN
+      };
+
+      dns.createZone(options, function(err, zone) {
+        assert.ifError(err);
+        zone.delete(done);
+      });
+    });
+
+    describe('changes', function() {
+      it('should create a change', function(done) {
+        var record = ZONE.record('srv', {
+          ttl: 3600,
+          name: DNS_DOMAIN,
+          data: '10 0 5222 127.0.0.1.'
+        });
+
+        ZONE.createChange({ add: record }, function(err, change) {
+          assert.ifError(err);
+
+          var addition = change.metadata.additions[0];
+          delete addition.kind;
+          assert.deepEqual(addition, record.toJSON());
+
+          done();
+        });
+      });
+
+      it('should get a list of changes', function(done) {
+        ZONE.getChanges(function(err, changes) {
+          assert.ifError(err);
+          assert(changes.length >= 0);
+          done();
+        });
+      });
+
+      it('should get metadata', function(done) {
+        ZONE.getChanges(function(err, changes) {
+          assert.ifError(err);
+
+          var change = changes[0];
+          var expectedMetadata = change.metadata;
+
+          change.getMetadata(function(err, metadata) {
+            assert.ifError(err);
+
+            delete metadata.status;
+            delete expectedMetadata.status;
+            assert.deepEqual(metadata, expectedMetadata);
+
+            done();
+          });
+        });
+      });
+    });
+  });
+
+  describe('Records', function() {
+    it('should return 0 or more records', function(done) {
+      ZONE.getRecords(function(err, records) {
+        assert.ifError(err);
+        assert(records.length >= 0);
+        done();
+      });
+    });
+
+    it('should replace records', function(done) {
```

Comment:
See the code in this test for where I went the most off-the-rails :train: from the original specs:

- `zone.record` taking a record type
- `zone.getRecords` taking a type
- `zone.replaceRecords` to delete records of a type and replace them with the ones provided

I added these to try to more easily solve a simple problem: "I got a new server and want to point my site at it" - Let me know if this is wacky.